File: 15-1/backend/schemas/access.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class AccessCheckInRequest(BaseModel):
    image: str = Field(..., description="Base64 encoded image")


class AccessResponse(BaseModel):
    id: int
    userId: int
    userName: str
    organizationType: str
    checkInTime: str
    similarity: Optional[float] = None
    status: str
    createdAt: str

    class Config:
        from_attributes = True


class AccessListResponse(BaseModel):
    total: int
    items: list[AccessResponse]


class AccessStatsItem(BaseModel):
    label: str
    count: int
    date: str
    firstCheckIn: float | None = None 

class AccessStatsResponse(BaseModel):
    period: str
    items: list[AccessStatsItem]

File: 15-1/frontend/src/api/access.ts
import apiClient from './client'

export interface AccessRequest {
  image: string
}

export interface AccessResponse {
  id: number
  userId: number
  userName: string
  organizationType: string
  checkInTime: string
  similarity?: number
  status: string
  createdAt: string
}

export interface AccessListResponse {
  total: number
  items: AccessResponse[]
}

export const checkIn = async (data: AccessRequest): Promise<AccessResponse> => {
  const response = await apiClient.post<AccessResponse>('/access/check-in', data)
  return response.data
}

export const getAccessHistory = async (limit = 50, offset = 0): Promise<AccessListResponse> => {
  const response = await apiClient.get<AccessListResponse>('/access/history', {
    params: { limit, offset },
  })
  return response.data
}

export interface AccessStatsItem {
  label: string
  count: number
  date: string
  firstAccessTime?: number
}

export interface AccessStatsResponse {  
  period: string
  items: AccessStatsItem[]
}

export const getAccessStats = async (period: 'minute' | 'hour' | 'day' | 'month' | 'year' = 'day'): Promise<AccessStatsResponse> => {
  const response = await apiClient.get<AccessStatsResponse>('/access/stats', {
    params: { period },
  })
  return response.data
}

